spec/core_ext/math/slope_spec.rb

```diff
@@ -0,0 +1,20 @@
+describe Math do
+  it ".slope_x_intercept" do
+    expect(described_class.slope_x_intercept(1, 1, 0)).to eq(1)
+    expect(described_class.slope_x_intercept(2, 1, 0)).to eq(2)
+    expect(described_class.slope_x_intercept(10, 1, 0)).to eq(10)
+    expect(described_class.slope_x_intercept(10, 2, 0)).to eq(5)
+  end
+
+  it ".slope_y_intercept" do
+    expect(described_class.slope_y_intercept(1, 1, 0)).to eq(1)
+    expect(described_class.slope_y_intercept(2, 1, 0)).to eq(2)
+    expect(described_class.slope_y_intercept(10, 1, 0)).to eq(10)
+    expect(described_class.slope_y_intercept(10, 2, 0)).to eq(20)
+  end
+
+  it ".slope" do
+    expect(described_class.slope([1.0, 2.0], [1.0, 2.0])).to eq([1.0, 0.0, 1.0])
+    expect(described_class.slope([1.0, 1.5], [1.0, 2.0])).to eq([2.0, -1.0, 1.0])
```

Comment:
I was looking at this yesterday and the output array consists of `slope`, `y-intercept` and `correlation coefficient`. We definitely use the `y-intercept` . The correlation coefficient is a measure of the strength of slope calculation based on the inputs. I haven't been able to find a place with it is being used yet. But, I can't say this it's not being used.